### PR TITLE
Propagate inputs info for ONNX and TFLite models

### PR DESCRIPTION
**Pull request by dkurt**
### Pull Request Readiness Checklist

Needed for generic applications such as benchmarking pipelines. So OpenCV can tell about the default input shapes specified in the models.

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/dnn/src/tflite/tflite_importer.cpp [hunk 1]
```diff
@@ -163,6 +163,8 @@ void TFLiteImporter::populateNet()
     CV_Assert(modelTensors);
     layouts.resize(modelTensors->size(), DATA_LAYOUT_UNKNOWN);
     size_t subgraph_inputs_size = subgraph_inputs->size();
+    std::vector<std::string> inputsNames;
+    std::vector<MatShape> inputsShapes;
```
Comment (asmorkalov) [inline modules/dnn/src/tflite/tflite_importer.cpp:167]: vector sizes are known here. It makes sense to define size and do not use push_back.

File: modules/dnn/src/net_impl.cpp [hunk 2]
```diff
@@ -1410,8 +1410,8 @@ void Net::Impl::setInput(InputArray blob, const String& name, double scalefactor
             const MatShape& inputShapeLimitation = netInputLayer.shapes[pin.oid];
             if (!inputShapeLimitation.empty())
             {
-                CV_CheckEQ(inputShapeLimitation.size(), blobShape.size(), "");
 #if 0  // TODO: DNNTestNetwork.MobileNet_SSD_Caffe_Different_Width_Height/0
+                CV_CheckEQ(inputShapeLimitation.size(), blobShape.size(), "");
```
Comment (asmorkalov) [inline modules/dnn/src/net_impl.cpp:1414]: It's not relevant change.

Comment (dkurt) [reply modules/dnn/src/net_impl.cpp:1414]: The following tests were failed:
```cpp
[  FAILED  ] Test_ONNX_layers.Convolution_variable_weight_bias/0, where GetParam() = OCV/CPU
[  FAILED  ] Test_ONNX_layers.Conv1d_variable_weight_bias/0, where GetParam() = OCV/CPU
```
```
C++ exception with description "OpenCV(4.7.0-dev) /home/dkurt/opencv/modules/dnn/src/net_impl.cpp:1413: error: (-2:Unspecified error) in function 'virtual void cv::dnn::dnn4_v20221220::Net::Impl::setInput(cv::InputArray, const String&, double, const Scalar&)'
>  (expected: 'inputShapeLimitation.size() == blobShape.size()'), where
>     'inputShapeLimitation.size()' is 1
> must be equal to
>     'blobShape.size()' is 2
" thrown in the test body.
```
The thing is that there is no limitation for the shapes - OpenCV networks are flexible for any valid input size,

File: modules/dnn/src/tflite/tflite_importer.cpp (hunk 1 above)
Comment (dkurt) [reply modules/dnn/src/tflite/tflite_importer.cpp:167]: Sure, thanks

File: modules/dnn/src/net_impl.cpp (hunk 2 above)
Comment (asmorkalov) [reply modules/dnn/src/net_impl.cpp:1414]: It makes sense to hide/remove `inputShapeLimitation` constant with the condition. It's abandoned constant with empty condition.

**Comment by dkurt** [comment]
Seems there is an unstable test: https://pullrequest.opencv.org/buildbot/builders/precommit_windows64/builds/101037
```
[ RUN      ] Conv3D.conv3d/14, where GetParam() = (GFLOPS=0.000, K=[1 x 1 x 1], IN={1, 8, 1, 10, 10}, OCN=8, G=8, P=(1, 1) x (1, 1) x (1, 1), BIAS, OCV/CPU)
C:\build\precommit_windows64\4.x\opencv\modules\ts\src\ts_perf.cpp(1975): error: Failed
Expected: PerfTestBody() doesn't throw an exception.
  Actual: it throws...
params    = (GFLOPS=0.000, K=[1 x 1 x 1], IN={1, 8, 1, 10, 10}, OCN=8, G=8, P=(1, 1) x (1, 1) x (1, 1), BIAS, OCV/CPU)
termination reason:  unhandled exception
bytesIn   =          0
bytesOut  =          0
samples   =          0 of 1
outliers  =          0
frequency =          0
[  FAILED  ] Conv3D.conv3d/14, where GetParam() = (GFLOPS=0.000, K=[1 x 1 x 1], IN={1, 8, 1, 10, 10}, OCN=8, G=8, P=(1, 1) x (1, 1) x (1, 1), BIAS, OCV/CPU) (3 ms)
```

File: modules/dnn/src/net_impl.cpp (hunk 2 above)
Comment (dkurt) [reply modules/dnn/src/net_impl.cpp:1414]: Decided to hide it under `#if 0` completely.

**Review by asmorkalov — APPROVED** [review]

**Pull request merged**